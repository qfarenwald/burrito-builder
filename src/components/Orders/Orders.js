import React from 'react';
import './Orders.css';
import { connect } from 'react-redux';

const Orders = ({ orders }) => {
  const orderEls = orders.map((order, i) => {
    return (
      <div className="order" key={i}>
        <h3>{order.name}</h3>
        <ul className="ingredient-list">
          {order.ingredients.map((ingredient, i) => {
            return <li key={i}>{ingredient}</li>
          })}
        </ul>
      </div>
    )
  });

  return (
    <section>
      { orderEls.length ? orderEls : <p>No orders yet!</p> }
    </section>
  )
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export default connect(mapStateToProps, null)(Orders);
