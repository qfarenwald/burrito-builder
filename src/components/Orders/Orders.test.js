import React from 'react';
import { Orders } from './Orders';
import { shallow } from 'enzyme';

describe('Orders', () => {
  it('should match the snapshot', () => {
    const mockOrders = [
      {
            "id": 1,
            "name": "Pat",
            "ingredients": [
                "beans",
                "lettuce",
                "carnitas",
                "queso fresco",
                "jalapeno"
            ]
        },
        {
            "id": 2,
            "name": "Sam",
            "ingredients": [
                "steak",
                "pico de gallo",
                "lettuce",
                "carnitas",
                "queso fresco",
                "jalapeno"
            ]
        },
    ]
    const wrapper = shallow(<Orders orders={mockOrders}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
