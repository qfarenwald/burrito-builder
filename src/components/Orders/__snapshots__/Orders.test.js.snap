// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="0"
  >
    <h3>
      Pat
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        beans
      </li>
      <li
        key="1"
      >
        lettuce
      </li>
      <li
        key="2"
      >
        carnitas
      </li>
      <li
        key="3"
      >
        queso fresco
      </li>
      <li
        key="4"
      >
        jalapeno
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="1"
  >
    <h3>
      Sam
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        steak
      </li>
      <li
        key="1"
      >
        pico de gallo
      </li>
      <li
        key="2"
      >
        lettuce
      </li>
      <li
        key="3"
      >
        carnitas
      </li>
      <li
        key="4"
      >
        queso fresco
      </li>
      <li
        key="5"
      >
        jalapeno
      </li>
    </ul>
  </div>
</section>
`;
